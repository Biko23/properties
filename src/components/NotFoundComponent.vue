<template>
    <div class="main-not-found">
        <center>
            <h4>Oops!! The page you are looking for might not be available</h4>
            <br/>
            <router-link :to="{name: 'Home'}">Go to Home</router-link>
        </center>
    </div>
</template>

<script>
    export default {
        name: 'NotFoundPage'
    }
</script>

<style scoped>
    .main-not-found {
        margin: 5em auto;
        display: flex;
        flex-direction: row;
        justify-content: center;
        align-items: center;
    }
</style>