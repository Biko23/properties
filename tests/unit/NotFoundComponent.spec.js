import { createLocalVue, shallowMount } from '@vue/test-utils'
import Vuetify from 'vuetify'
import VueRouter from 'vue-router'
import NotFoundComponent from '@/components/NotFoundComponent.vue'

describe('NotFoundComponent.vue', () => {
    const localVue = createLocalVue()
    localVue.use(VueRouter)
    let vuetify
    let router

    beforeEach(() => {
        vuetify = new Vuetify()
        router = new VueRouter()
    })

    const mountFunction = options => {
        return shallowMount(NotFoundComponent, {
            localVue,
            vuetify,
            router,
            ...options
        })
    }

    it('should have a match snapshot', () => {
        const wrapper = mountFunction()
        expect(wrapper.html()).toMatchSnapshot()
    })
})

