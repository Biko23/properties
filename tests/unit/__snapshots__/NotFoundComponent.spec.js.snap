// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundComponent.vue should have a match snapshot 1`] = `
<div class="main-not-found">
  <center>
    <h4>Oops!! The page you are looking for might not be available</h4> <br>
    <router-link-stub to="[object Object]" tag="a" ariacurrentvalue="page" event="click">Go to Home</router-link-stub>
  </center>
</div>
`;
